File: Assets/ApertureCutsceneEditor/Components/Actions/Action_AudioProperties.js
#pragma strict
@script AddComponentMenu("Cutscene/Actions/Audio Properties")

@script ExecuteInEditMode()

@CutsceneEventOverrideNameAttribute("Audio Properties")

class Action_AudioProperties extends CutsceneAction {
	
	//#if !UNITY_FLASH
	@CutsceneActionAttribute("Set Volume")
	//#endif
	function SetVolume(value:float) {
		if (!audio) return;
		audio.volume = value;
	}
	
	//#if !UNITY_FLASH
	@CutsceneActionAttribute("Set Pitch")
	//#endif
	function SetPitch(value:float) {
		if (!audio) return;
		audio.pitch = value;
	}
	
	//#if !UNITY_FLASH
	@CutsceneActionAttribute("Set Time")
	//#endif
	function SetTime(value:float) {
		if (!audio) return;
		audio.time = value;
	}
}

File: CutsceneEditor/CutsceneEditor/CutsceneEditor/Components/Actions/Action_AudioProperties.cs
using UnityEngine;

[AddComponentMenu("Cutscene/Actions/Audio Properties")]

[ExecuteInEditMode()]

[CutsceneEventOverrideNameAttribute("Audio Properties")]

class Action_AudioProperties : CutsceneAction {
	
	[CutsceneActionAttribute("Set Volume")]
	public void SetVolume(float value) {
		if (!audio) return;
		audio.volume = value;
	}
	
	[CutsceneActionAttribute("Set Pitch")]
	public void SetPitch(float value) {
		if (!audio) return;
		audio.pitch = value;
	}
	
	[CutsceneActionAttribute("Set Time")]
	public void SetTime(float value) {
		if (!audio) return;
		audio.time = value;
	}
}

